.github/workflows/build-test.yml

```diff
@@ -895,14 +895,41 @@ jobs:
         path: ./benchmarks/datasize/**
         name: benchmark-datasize
 
+  # Docs job
+
+  doc:
```

Comment:
@sffc Should we add it to the check job or use a separate job?

We could also lump it in with the clippy job, tbh, the reason we're running this is to get lint results.

A slight problem is that when it's a separate job it needs to rerun the metadata build (whereas clippy/check already have done so)

Reply:
We already run three separate lint jobs: `check`, `fmt`, and `clippy`. I think it's best to add this as a fourth and parallelize to ensure fast results rather than optimize for total time, as the actual build and benchmarks probably account for the majority of that anyway.

Reply:
Sounds good.

Reply:
My worry is that mostly there's a limit to the number of simultaneous jobs, so reducing total time spent on CI is still useful and it can be done by merging specific kinds of jobs.

I wish GHA allowed forking jobs, it would simplify a lot of this.

Reply:
We can upload the `target` and `.cargo` directories as artifacts and use them in subsequent workflows, which is somewhat like a fork.